java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/protocol/match/MatchField.java

```diff
@@ -31,15 +31,24 @@
 import org.projectfloodlight.openflow.types.VxlanNI;
 import org.projectfloodlight.openflow.types.VFI;
 
+import java.util.Collections;
+import java.util.Set;
+import java.util.HashSet;
+
 public class MatchField<F extends OFValueType<F>> {
     private final String name;
     public final MatchFields id;
-    private final Prerequisite<?>[] prerequisites;
+    private final Set<Prerequisite<?>> prerequisites;
```

Comment:
We should use a Guava [`ImmutableSet`](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/ImmutableSet.html) here, then we don't need to do the defensive wrapper in the getter, and also it has a direct factory that you can use below:
```
  this.prerequisites = ImmutableSet.copyOf(prerequisites);
```

(Another advantage is that ImmutableSet is order-preserving, while HashMap will create an arbitrary iteration order).